idaes/apps/grid_integration/tracker.py

```diff
@@ -0,0 +1,261 @@
+import pandas as pd
+import pyomo.environ as pyo
+from collections import deque
+import os
+
+class Tracker:
+
+    def __init__(self, tracking_model_class, n_tracking_hour, solver, **kwarg):
+
+        '''
+        Initializes the tracker object.
+
+        Arguments:
+            tracking_model_class: the model object class for tracking
+            n_tracking_hour: number of implemented hours after each solve
+            solver: a Pyomo mathematical programming solver object
+            **kwarg: necessary arguments to initialize the selected model object
+
+        Returns:
+            None
+        '''
+
+        # create an instance
+        self.tracking_model_object = tracking_model_class(**kwarg)
+
+        # add flowsheet to model
+        self.model = pyo.ConcreteModel()
+        self.model.fs = pyo.Block()
```

Comment:
self.model.fs = self.tracking_model_object.create_model()

Reply:
This avoids need to call `transfer_attributes_form`

Reply:
Bidder can do the same thing. Rule calls create model.

Reply:
self.model.fs[t] = self.tracking_model_object.create_model()

Reply:
See discussion from John.

Reply:
Suggestion from @jsiirola:
```
model = ConcreteModel()
model.fs = Block(model.TIME)
model.fs[1] = self.tracker_model.create_model()
# ^^^ this works, but Pyomo will call transfer_attributes_from under the hood

# ...alternatively:

self.tracker_model.populate_{something}(model.fs[1])

# If you want to take an arbitrary user model

class Wrapper():
    def populate_model(b):
        b.user_model = create_user_model()
```

Reply:
@jalving See this

Reply:
@jsiirola Is going to draft up a clean implementation for @jalving and possibly @xiangao1